src/_pytest/skipping.py

```diff
@@ -101,6 +102,14 @@ def evaluate_condition(item: Item, mark: Mark, condition: object) -> Tuple[bool,
             "platform": platform,
             "config": item.config,
         }
+        for dictionary in item.ihook.pytest_markeval_namespace(config=item.config):
```

Comment:
```suggestion
        for dictionary in reversed(item.ihook.pytest_markeval_namespace(config=item.config)):
```
Is this an acceptable solution?